### ISSUE TITLE
444 unit test failures on AARCH64 build of 2.12.0-rc0

### ISSUE THREAD

**Issue by elfringham** [tensorflow/tensorflow#59765]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

Yes

### Source

source

### Tensorflow Version

2.12.0-rc0

### Custom Code

No

### OS Platform and Distribution

CentOS 7

### Mobile device

n/a

### Python version

3.8.10

### Bazel version

5.3.0

### GCC/Compiler version

10.2.1

### CUDA/cuDNN version

n/a

### GPU model and memory

n/a

### Current Behaviour?

```shell
Executed 3688 out of 3688 tests: 3244 tests pass and 444 fail locally.
```

### Standalone code to reproduce the issue

```shell
bazel test --test_timeout=300,500,-1,-1 \
              --flaky_test_attempts=3 \
              --test_output=all \
              --cache_test_results=no \
              --config=nonccl \
              --config=mkl_aarch64_threadpool \
              --copt="-mtune=generic" \
              --copt="-march=armv8-a" \
              --copt="-O3" \
              --verbose_failures \
              --test_env=TF_ENABLE_ONEDNN_OPTS=1 \
              --build_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-no_oss_py38,-no_oss_py39,-no_oss_py310 \
              --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only,-no_aarch64,-no_oss_py38,-no_oss_py39,-no_oss_py310 \
              --build_tests_only \
              -- ${DEFAULT_BAZEL_TARGETS} \
              -//tensorflow/lite/... -//tensorflow/compiler/mlir/lite/tests:const-fold.mlir.test -//tensorflow/core/distributed_runtime/integration_test:c_api_session_coordination_test_cpu -//tensorflow/core/platform:ram_file_system_test -//tensorflow/python/client:session_list_devices_test -//tensorflow/python/compiler/xla:xla_test_cpu -//tensorflow/python/data/experimental/kernel_tests:checkpoint_input_pipeline_hook_test -//tensorflow/python/data/kernel_tests:iterator_test_cpu -//tensorflow/python/distribute:parameter_server_strategy_test_cpu -//tensorflow/python/distribute:parameter_server_strategy_test_2gpu -//tensorflow/python/kernel_tests/linalg:matrix_triangular_solve_op_test -//tensorflow/python/training:server_lib_test -//tensorflow/tsl/framework/convolution:spatial_convolutions_test \
              -//tensorflow/compiler/mlir/tfr/examples/mnist:mnist_ops_test -//tensorflow/core/grappler/optimizers:auto_mixed_precision_test_cpu -//tensorflow/core/grappler/optimizers:remapper_test_cpu -//tensorflow/python/kernel_tests/nn_ops:atrous_conv2d_test_cpu -//tensorflow/python/kernel_tests/nn_ops:conv_ops_test_cpu
```

### Relevant log output

```shell
No useful log output, tests immediately segfault.
```
</details>

**Comment by elfringham** [comment]
The start of these failures was bisected to https://github.com/tensorflow/tensorflow/commit/84f40925e929d05e72ab9234e53c729224e3af38

**Comment by elfringham** [comment]
Changing the installed version of the Python protobuf package did not make a difference, 3.20.1, 3.20.3, 4.21.9, 4.21.12 all the same failure.

**Comment by elfringham** [comment]
This failure is seen inside the manylinux2014_aarch64 docker container which uses gcc 10.2.1. I tested updating gcc to 10.4.0 and the tests then passed. So this issue seems to be an interaction between protobuf and gcc.

**Comment by elfringham** [comment]
Possible routes to an expected resolution are
1) Create a new docker container based on manylinux2014_aarch64 container, but with gcc updated to 10.4.0
2) Move build onto new docker container built by Linaro that has been developed to allow building with CXX11 Dual ABI enabled.

**Comment by elfringham** [comment]
Solution 1) above is dead end work, it would likely only be used for the 2.12 release branch.
2) needs work to update the Github actions and co-ordinating a release with tensorflow-io as well as work to the Linaro CI jobs.

**Comment by SuryanarayanaY** [comment]
Hi @elfringham ,

Thanks for the tests and inputs. As build configurations for Tf 2.12v not updated in documentation shall we ensure GCC version to be 10.4.0 as it seems anything below this not working as per your tests?

**Comment by elfringham** [comment]
Hi @SuryanarayanaY well apart from suspecting that gcc 10.3.x would also work, but that is not tested.
Unfortunately when I suggested this I had not realised that there was not a suitably patched release of gcc-10 > 10.2.1 that would be easily usable in CentOS 7. So that makes this option much less desirable. I could try gcc-11 but that worries me as despite there being a source package for it, there are no matching AARCH64 binaries, and of course it has not been validated for building TensorFlow.
The test I did with gcc 10.4.0 was by installing it from conda and produced binaries that were not compatible with manylinux2014_aarch64.
